packages/widget-v2/src/modals/TokenAndChainSelectorModal/TokenAndChainSelectorModal.tsx

```diff
@@ -187,15 +187,22 @@ export const TokenAndChainSelectorModal = createModal(
 
     const filteredChains = useMemo(() => {
       if (!selectedGroup || !chains) return;
+
+      const seenChainIds = new Set<string>();
+
       const resChains = selectedGroup.assets
         .map((asset) => {
-          const c = chains.find((c) => c.chainID === asset.chainID);
-          return {
-            ...c,
-            asset,
-          };
+          const chain = chains.find((c) => c.chainID === asset.chainID);
+          return chain ? { ...chain, asset } : null;
         })
-        .filter((c) => c) as ChainWithAsset[];
+        .filter((chain) => {
+          if (!chain) return false;
+          if (seenChainIds.has(chain.chainID)) return false;
+
+          seenChainIds.add(chain.chainID);
+          return true;
+        }) as ChainWithAsset[];
+
```

Comment:
I need to think about how to handle different assets with the same recommended symbol and chain ID

Reply:
this is naive and just takes whichever one with the same chainID is first in the aray